### PR TITLE
[Function] support vec bitmap_count

### PR DESCRIPTION
**Pull request by zbtzbtzbt**
## Proposed changes

**Implement vec function bitmap_xx_count:**
- [x] bitmap_and_count
- [x] bitmap_or_count
- [x] bitmap_xor_count

**Todo:**

1. there are a lot of similar codes, only diff is operator `&` `|` `^`, so these codes can be implemented in a more concise way.
2. some vec bitmap function lack unit test, we need add it later.

**Question:**

1. can we use SIMD to accelerate these function?

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)
- [ ] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have created an issue on (Fix #ISSUE) and described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/functions/function_bitmap.cpp [hunk 1]
```diff
@@ -319,12 +319,84 @@ struct BitmapToString {
     }
 };
 
+struct NameBitmapAndCount{
```
Comment (HappenLee) [inline be/src/vec/functions/function_bitmap.cpp:322]: `NameBitmapAndCount {`空格

File: be/src/vec/functions/function_bitmap.cpp [hunk 2]
```diff
@@ -319,12 +319,84 @@ struct BitmapToString {
...
+    using ResultDataType = DataTypeInt64;
+    using T0 = typename LeftDataType::FieldType;
+    using T1 = typename RightDataType::FieldType;
+    using TData = std::vector<BitmapValue>;
+    using ResTData = typename ColumnVector<Int64>::Container;
+    
+    static Status vector_vector(const TData& lvec, const TData& rvec, ResTData& res) {
+        size_t size = lvec.size();
+        for (size_t i = 0; i < size; ++i) {
+            BitmapValue val = lvec[i];
```
Comment (HappenLee) [inline be/src/vec/functions/function_bitmap.cpp:336]: 使用引用

File: be/src/vec/functions/function_bitmap.cpp [hunk 3]
```diff
@@ -319,12 +319,84 @@ struct BitmapToString {
...
```
Comment (HappenLee) [inline be/src/vec/functions/function_bitmap.cpp:358]: 同上

**Review by HappenLee — COMMENTED** [review]
有些问题，再check一下

File: be/src/vec/functions/function_bitmap.cpp (hunk 2 above)
Comment (zbtzbtzbt) [reply be/src/vec/functions/function_bitmap.cpp:336]: ok

Comment (zbtzbtzbt) [reply be/src/vec/functions/function_bitmap.cpp:336]: 我理解你的意思是用引用避免一次拷贝构造，效率更高。但lvec是const类型，val作为它的引用的话，后续继续操作会编译错误，所以不能这样优化。

File: be/src/vec/functions/function_bitmap.cpp (hunk 1 above)
Comment (zbtzbtzbt) [reply be/src/vec/functions/function_bitmap.cpp:322]: ok

File: be/src/vec/functions/function_bitmap.cpp (hunk 2 above)
Comment (HappenLee) [reply be/src/vec/functions/function_bitmap.cpp:336]: 那你别传const引用？

Comment (zbtzbtzbt) [reply be/src/vec/functions/function_bitmap.cpp]: ok

**Comment by zbtzbtzbt** [comment]
1.使用BitmapValue&避免大对象的拷贝导致的开销 2.int改为size_t避免大值溢出导致的不符合预期的for循环

**Pull request merged**